medium-tests/src/test/java/mediumtest/analysis/AutomaticAnalysisEnablementMediumTests.java

```diff
@@ -0,0 +1,115 @@
+import static org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.BackendCapability.EMBEDDED_SERVER;
+
+class AutomaticAnalysisEnablementMediumTests {
+
+  @RegisterExtension
+  SonarLintLogTester logTester = new SonarLintLogTester(true);
+
+  private final Gson gson = new Gson();
+
+  @SonarLintTest
+  void it_should_enable_automatic_analysis(SonarLintTestHarness harness) throws IOException, InterruptedException {
+    var backend = harness.newBackend()
+      .withBackendCapability(EMBEDDED_SERVER)
+      .start();
+
+    var response = executeAutomaticAnalysisEnablementRequest(backend, "enabled=true");
+    
+    await().atMost(2, TimeUnit.SECONDS).untilAsserted(() -> assertThat(response.statusCode()).isEqualTo(200));
+    var automaticAnalysisEnablementResponse = gson.fromJson(response.body(), AutomaticAnalysisEnablementRequestHandler.AutomaticAnalysisEnablementResponse.class);
+    assertThat(automaticAnalysisEnablementResponse.success()).isTrue();
+    assertThat(automaticAnalysisEnablementResponse.message()).contains("enabled");
+  }
+
+  @SonarLintTest
+  void it_should_disable_automatic_analysis(SonarLintTestHarness harness) throws IOException, InterruptedException {
+    var backend = harness.newBackend()
+      .withBackendCapability(EMBEDDED_SERVER)
+      .start();
+
+    var response = executeAutomaticAnalysisEnablementRequest(backend, "enabled=false");
+
+    await().atMost(2, TimeUnit.SECONDS).untilAsserted(() -> assertThat(response.statusCode()).isEqualTo(200));
+    var automaticAnalysisEnablementResponse = gson.fromJson(response.body(), AutomaticAnalysisEnablementRequestHandler.AutomaticAnalysisEnablementResponse.class);
+    assertThat(automaticAnalysisEnablementResponse.success()).isTrue();
+    assertThat(automaticAnalysisEnablementResponse.message()).contains("disabled");
+  }
+
+  @SonarLintTest
+  void it_should_return_bad_request_for_missing_parameter(SonarLintTestHarness harness) throws IOException, InterruptedException {
+    var backend = harness.newBackend()
+      .withBackendCapability(EMBEDDED_SERVER)
+      .start();
+
+    var response = executeAutomaticAnalysisEnablementRequest(backend, "invalid=param");
+
+    await().atMost(2, TimeUnit.SECONDS).untilAsserted(() -> assertThat(response.statusCode()).isEqualTo(400));
+  }
+
+  @SonarLintTest
+  void it_should_return_bad_request_for_get_request(SonarLintTestHarness harness) throws IOException, InterruptedException {
+    var backend = harness.newBackend()
+      .withBackendCapability(EMBEDDED_SERVER)
+      .start();
+
+    await().atMost(2, TimeUnit.SECONDS).untilAsserted(() -> 
+      assertThat(backend.getEmbeddedServerPort()).isGreaterThan(0));
+
+    var request = HttpRequest.newBuilder()
+      .uri(URI.create("http://localhost:" + backend.getEmbeddedServerPort() + "/sonarlint/api/analysis/automatic/config"))
+      .header("Origin", "http://localhost")
+      .header("Content-Type", "application/json")
+      .GET()
+      .build();
```

Comment:
Maybe you could factorize this together with `executeAutomaticAnalysisEnablementRequest`? E.g. with a different parameter